nexus/db-model/src/inventory.rs

```diff
@@ -881,16 +882,65 @@ impl InvPhysicalDisk {
     }
 }
 
-impl From<InvPhysicalDisk> for nexus_types::inventory::PhysicalDisk {
-    fn from(disk: InvPhysicalDisk) -> Self {
-        Self {
-            identity: omicron_common::disk::DiskIdentity {
-                vendor: disk.vendor,
-                serial: disk.serial,
-                model: disk.model,
-            },
-            variant: disk.variant.into(),
-            slot: disk.slot,
+/// See [`nexus_types::inventory::PhysicalDiskFirmware::Nvme`].
+#[derive(Queryable, Clone, Debug, Selectable, Insertable)]
+#[diesel(table_name = inv_nvme_disk_firmware)]
+pub struct InvNvmeDiskFirmware {
+    pub inv_collection_id: DbTypedUuid<CollectionKind>,
+    pub sled_id: DbTypedUuid<SledKind>,
+    pub slot: i64,
+    pub active_slot: SqlU8,
+    pub next_active_slot: Option<SqlU8>,
+    pub number_of_slots: SqlU8,
+    pub slot1_is_read_only: bool,
+    pub slot_firmware_versions: Vec<Option<String>>,
```

Comment:
Nitpick: Up to you, but if it's a constraint that the `slot_firmware_version` values be less than 8 characters (are there other constraints?) we might want to make a newtype wrapper around `String`, validating that we don't pass around invalid data here (e.g., strings greater than 8 characters)

Reply:
This should be fixed now by introducing a bunch of checks in the `InvNvmeDiskFirmware` type itself.